docs/research-branch-policy.md

```diff
@@ -0,0 +1,57 @@
+
+# Git branch strategies
+
+There are different strategies around git branching:
+
+- [Release-based branching](https://medium.com/@pooyagohardani/git-release-candidate-branches-strategy-balancing-quality-and-agility-in-software-development-21cc00842e03)
+- [Feature branching](https://martinfowler.com/articles/branching-patterns.html)
+- [Git flow](https://nvie.com/posts/a-successful-git-branching-model/)
+- [GitHub flow](https://docs.github.com/en/get-started/using-github/github-flow)
+- [Task branching](https://unity.com/how-to/devops-task-branch-workflow)
+- [Trunk-based development](https://trunkbaseddevelopment.com/)
+- Probably others!
+
+(More information on some of these including a comparison chart is available in [Git Branching Strategies for DevOps: Best Practices for Collaboration](https://dev.to/angelotheman/git-branching-strategies-for-devops-best-practices-for-collaboration-35l8) by Angel Oduro-Temeng Twumasi on dev.to.)
+
+The CLI repository [instructlab/instructlab](https://github.com/instructlab/instructlab) has a [Release Strategy devdoc](https://github.com/instructlab/instructlab/blob/main/docs/release-strategy.md) that dictates a release-based branching strategy. We do not have defined branching strategies for the other repos.
+
+We should probably decide on explicit branching policies for each of the core repositories in the `instructlab.` This will set clear expectations for our ways of working as a community project.
+
+# Research branch policy proposal
+
+Regardless of the branching strategy of the repo, we propose allowing a `release` branch on each of the core `instructlab`. This branch would not be subject to the same PR review guidelines as the rest of the repository. The goal of this branch is to enable rapid, multi-person collaborative development to get the code to a state that can then go through a more stringent code review process that adheres to the project's standards:
+
+- The `research` branch in each repository that has one will have named maintainers. Maintainer membership will be based on [the InstructLab maintainer policy already in place](https://github.com/instructlab/community/blob/main/GOVERNANCE.md#project-maintainers-overview).
+- Research-oriented PRs must be made against the `research` branch.
+- `research` branch maintainers will have the freedom to determine their PR merge policy. It will be documented in their branch's README or another easily-discoverable location of their choosing.
+- (Need to address CI in this policy. Not sure how or what is fair.)
+- When the `research` branch is in a state the maintainers are confident is good enough to attempt merging to main, they will coordinate with the repository core maintainers to start a review process.
```

Comment:
This makes me very nervous. I'm concerned that we'll end up with an enormous amount of low-quality code being proposed to main through PRs. How can we ensure this doesn't happen? Ideally, we will need small batches instead of trying to merge the entire `research` branch in `main`.

I think this is a very tedious process because we will likely not take every commit from `research` and push it into `main`. Some of that history will need to be rewritten to meet `main` quality standards. This sounds like a significant amount of effort for whoever will be responsible for pushing code from `research` into `main`.

I'd like to avoid a situation where `research` moves so quickly that merging into `main` cannot keep up without lowering the quality standards of `main`, thereby putting pressure on `main` to merge more rapidly.

Reply:
If i said s/research/devel on the name of the branch, would that change anything for you?

Reply:
No, same problem.